tls/s2n_security_policies.c

```diff
@@ -613,23 +619,45 @@ int s2n_find_security_policy_from_version(const char *version, const struct s2n_
     S2N_ERROR(S2N_ERR_INVALID_SECURITY_POLICY);
 }
 
+static S2N_RESULT s2n_validate_security_policy_choice(const struct s2n_security_policy *policy) {
+    ENSURE_REF(policy);
+    ENSURE_REF(policy->cipher_preferences);
+    ENSURE_REF(policy->kem_preferences);
+    ENSURE_REF(policy->signature_preferences);
+    ENSURE_REF(policy->ecc_preferences);
+
+    if (!s2n_pq_is_enabled()) {
```

Comment:
I think you should add a comment here along the lines of `PQ is automatically disabled if FIPS mode is detected.`